File: Examples/CSharp/Articles/FindCellsWithSpecificStyle.cs
using System.IO;
using System;
using Aspose.Cells;

namespace Aspose.Cells.Examples.CSharp.Articles
{
    public class FindCellsWithSpecificStyle
    {
        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            //Output directory
            string outputDir = RunExamples.Get_OutputDirectory();

            Workbook workbook = new Workbook(sourceDir + "sampleFindCellsWithSpecificStyle.xlsx");

            Worksheet worksheet = workbook.Worksheets[0];

            // Access the style of cell A1
            Style style = worksheet.Cells["A1"].GetStyle();

            // Specify the style for searching
            FindOptions options = new FindOptions();
            options.Style = style;

            Cell nextCell = null;

            do
            {
                // Find the cell that has a style of cell A1
                nextCell = worksheet.Cells.Find(null, nextCell, options);
                if (nextCell == null)
                    break;
                // Change the text of the cell
                nextCell.PutValue("Found");

            } while (true);

            workbook.Save(outputDir + "outputFindCellsWithSpecificStyle.xlsx");

            Console.WriteLine("FindCellsWithSpecificStyle executed successfully.\r\n");
        }
    }
}

File: Examples/VisualBasic/Articles/FindCellsWithSpecificStyle.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Cells

Namespace Articles
    Public Class FindCellsWithSpecificStyle
        Public Shared Sub Run()
            ' ExStart:1
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            Dim filePath As String = dataDir & "TestBook.xlsx"

            Dim workbook As New Workbook(filePath)

            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Access the style of cell A1
            Dim style As Style = worksheet.Cells("A1").GetStyle()

            ' Specify the style for searching
            Dim options As New FindOptions()
            options.Style = style

            Dim nextCell As Cell = Nothing

            Do
                ' Find the cell that has a style of cell A1
                nextCell = worksheet.Cells.Find(Nothing, nextCell, options)

                If nextCell Is Nothing Then
                    Exit Do
                End If

                ' Change the text of the cell
                nextCell.PutValue("Found")

            Loop While True

            workbook.Save(dataDir & "output.xlsx")
            ' ExEnd:1


        End Sub
    End Class
End Namespace